libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp

```diff
@@ -664,11 +673,18 @@ void ESP8266WiFiSTAClass::_smartConfigCallback(uint32_t st, void* result) {
     sc_status status = (sc_status) st;
     if(status == SC_STATUS_LINK) {
         station_config* sta_conf = reinterpret_cast<station_config*>(result);
-
-        wifi_station_set_config(sta_conf);
+        if(_scDecryptFn) {
+            char decryptedPass[64] = "";
+            _scDecryptFn(decryptedPass, (const char *)sta_conf->password);
+            memcpy(sta_conf->password,decryptedPass, 64);   

```

Comment:
Allocation of intermediate buffer `decryptedPass` would be obsoleted by providing buffer size to _scDecryptFn, done like  this it's particularly prone to malicious buffer overruns, or is it?